### PR TITLE
Adult child vax

### PR DESCRIPTION
**Pull request by ruthmccabe**
Changes to vaccination as discussed on Friday 18/10:
- fixing bugs in assessing targets 
- child and adult vax campaigns separate
- no vax uptake parameter

### CODE REVIEW

**Review by lwhittles — CHANGES_REQUESTED** [review]
Looks great! Have left a couple of suggestions to simplify the code but looks like it's all working as expected. 🥇

File: R/dust.R [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Generated by dust (version 0.15.3) - do not edit
+## Generated by dust (version 0.15.2) - do not edit
```
[the remark this answers is not in the document]
Comment (ruthmccabe) [reply R/dust.R]: done

File: inst/odin/model-targeted-vax.R [hunk 2]
```diff
@@ -125,63 +187,122 @@ n_vaccination_t_Ea[, ] <- 0
...
-## allocate 2nd doses
-n_vaccination_t_S[, 3] <- if (sum(n_eligible_for_dose2[]) == 0) 0 else
-  min(floor((daily_doses_t[3] * S[i, 3] *
-               prioritisation_strategy[i, prioritisation_step_2nd_dose] *
-               vaccine_uptake[i]) / sum(n_eligible_for_dose2[])),
+## for the boundary case do an extra check that we haven't gone over the number of people in each compartment 
+
+n_vaccination_t_S[4,2] <- if(n_vaccination_t_S[4,2]>S[4,2]) 
+  S[4,2] else
+  n_vaccination_t_S[4,2]
```
[the remark this answers is not in the document]
Comment (ruthmccabe) [reply inst/odin/model-targeted-vax.R:268]: loooool this is so much nicer, love that my brain didn't go there at all

File: inst/odin/model-targeted-vax.R [hunk 3]
```diff
@@ -14,108 +14,170 @@ update(time) <- (step + 1) * dt
...
+## children
+## 1st doses
+## if you have a 2nd dose this implies you also have had a 1st dose so account
+## for this in the 1st dose target - this now isn't strictly relevant for children but leaving it in in case we do expand this to 2 doses in future
+target_met_children_t[, 3] <-
+  if ((sum(N[i, 3:4])*children_ind_raw[i]) > # number of vaxxed kids in relevant compartments 
+      round(prioritisation_strategy_children[i, prioritisation_step_1st_dose_children] * #children_ind_raw[i] *
+            #vaccination_coverage_target_1st_dose_children_prop * 
+            sum(N[i, 1:4])))
+    1 else 0
```
[the remark this answers is not in the document]
Comment (ruthmccabe) [reply inst/odin/model-targeted-vax.R]: done

**Comment by ruthmccabe** [comment]
@lwhittles ready for you again when you have the chance!

**Review by lwhittles — APPROVED** [review]

**Pull request merged**